### PR TITLE
[SPARK-32622][SQL][TEST] Add case-sensitivity test for ORC predicate pushdown

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

During working on SPARK-25557, we found that ORC predicate pushdown doesn't have case-sensitivity test. This PR proposes to add case-sensitivity test for ORC predicate pushdown.

### Why are the changes needed?

Increasing test coverage for ORC predicate pushdown.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Pass Jenkins tests.

### CODE REVIEW

**Comment by viirya** [comment]
cc @HyukjinKwon @dongjoon-hyun

File: sql/core/v1.2/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilterSuite.scala [hunk 1]
```diff
@@ -513,5 +513,98 @@ class OrcFilterSuite extends OrcTest with SharedSparkSession {
...
+          sql(
+            s"""
+               |CREATE TABLE $tableName (a LONG) USING ORC LOCATION '$tableDir2'
+             """.stripMargin)
+
+          checkAnswer(sql(s"select a from $tableName"), (0 until count).map(c => Row(c - 1)))
+
+          val actual = stripSparkFilter(sql(s"select a from $tableName where a < 0"))
+          // TODO: ORC predicate pushdown should work under case-insensitive analysis.
+          // assert(actual.count() == 1)
```
Comment (viirya) [inline sql/core/v1.2/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilterSuite.scala:590]: @HyukjinKwon @dongjoon-hyun I think this is a bug. I found this when I added test for #29412.

Comment (viirya) [reply sql/core/v1.2/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilterSuite.scala:590]: This was already in Spark ORC datasource before SPARK-25557. Thus it's not a regression caused by nested predicate pushdown. So I will address this in a new ticket and PR.

Comment (dongjoon-hyun) [reply sql/core/v1.2/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilterSuite.scala:590]: Can we have this case on `branch-3.0`, too?

Comment (viirya) [reply sql/core/v1.2/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilterSuite.scala:590]: Yes, this should be fixed in branch-3.0 too.

Comment (viirya) [reply sql/core/v1.2/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilterSuite.scala:590]: I use original SPARK-25557 as PR title now. If we want to backport this test to branch-3.0, should I create a new JIRA ticket for this?

Comment (dongjoon-hyun) [reply sql/core/v1.2/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilterSuite.scala:590]: Yes, please~

**Comment by viirya** [comment]
Updated the JIRA ticket number.

**Comment by viirya** [comment]
retest this please

**Comment by viirya** [comment]
@dongjoon-hyun @HyukjinKwon Any more comment for this test?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @viirya .
Merged to master/3.0.

**Comment by HyukjinKwon** [comment]
Late LGTM too. Thanks @viirya and @dongjoon-hyun.

**Pull request closed without merging**